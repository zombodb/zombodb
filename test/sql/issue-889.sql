CREATE TABLE issue889
(
    id       SERIAL8 NOT NULL PRIMARY KEY,
    name     text    NOT NULL,
    testdate date,
    testval  varchar
);
CREATE INDEX idxissue889
    ON issue889 USING zombodb ((issue889.*)) WITH (field_lists='my_list=[testdate,testval]');

INSERT INTO issue889 (name, testdate, testval)
VALUES ('pretzels', '2024-04-01', '72');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('cheese', '2024-04-02', '82');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('cocktail weenies', '2024-04-01', 'none');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('buffalo wings', '2024-04-01', '2024-04-09');

SELECT *
FROM issue889
WHERE issue889 ==> 'my_list="2024-04-02"'
ORDER BY id;


WITH matches as MATERIALIZED (SELECT *
                              FROM issue889
                              WHERE id = ANY ('{"1","2","3"}')),
     highlights AS MATERIALIZED (SELECT (zdb.highlight_document(
             'issue889'::regclass,
             json_build_object('name', name, 'testdate', testdate, 'testval', testval),
             'my_list="2024-04-02"'::TEXT)).*,
                                        id as primary_key
                                 FROM matches)
SELECT *
FROM highlights
ORDER BY "primary_key", "field_name", "position";

WITH matches as MATERIALIZED (SELECT *
                              FROM issue889
                              WHERE id = ANY ('{"1","2","3","4"}')),
     highlights AS MATERIALIZED (SELECT (zdb.highlight_document(
             'issue889'::regclass,
             json_build_object('name', name, 'testdate', testdate, 'testval', testval),
             'my_list="2024-04-02" OR my_list="2024-04-09" OR testdate="2024-04-01"')).*,
                                        id as primary_key
                                 FROM matches)
SELECT *
FROM highlights
ORDER BY "primary_key", "field_name", "position";

DROP TABLE issue889 CASCADE;
