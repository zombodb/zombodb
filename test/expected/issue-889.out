CREATE TABLE issue889
(
    id       SERIAL8 NOT NULL PRIMARY KEY,
    name     text    NOT NULL,
    testdate date,
    testval  varchar
);
CREATE INDEX idxissue889
    ON issue889 USING zombodb ((issue889.*)) WITH (field_lists='my_list=[testdate,testval]');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('pretzels', '2024-04-01', '72');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('cheese', '2024-04-02', '82');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('cocktail weenies', '2024-04-01', 'none');
INSERT INTO issue889 (name, testdate, testval)
VALUES ('buffalo wings', '2024-04-01', '2024-04-09');
SELECT *
FROM issue889
WHERE issue889 ==> 'my_list="2024-04-02"'
ORDER BY id;
 id |  name  |  testdate  | testval 
----+--------+------------+---------
  2 | cheese | 04-02-2024 | 82
(1 row)

WITH matches as MATERIALIZED (SELECT *
                              FROM issue889
                              WHERE id = ANY ('{"1","2","3"}')),
     highlights AS MATERIALIZED (SELECT (zdb.highlight_document(
             'issue889'::regclass,
             json_build_object('name', name, 'testdate', testdate, 'testval', testval),
             'my_list="2024-04-02"'::TEXT)).*,
                                        id as primary_key
                                 FROM matches)
SELECT *
FROM highlights
ORDER BY "primary_key", "field_name", "position";
 field_name | array_index |    term    |  type  | position | start_offset | end_offset |     query_clause      | primary_key 
------------+-------------+------------+--------+----------+--------------+------------+-----------------------+-------------
 testdate   |           0 | 2024-04-02 | <DATE> |        1 |            0 |          0 | testdate="2024-04-02" |           2
(1 row)

WITH matches as MATERIALIZED (SELECT *
                              FROM issue889
                              WHERE id = ANY ('{"1","2","3","4"}')),
     highlights AS MATERIALIZED (SELECT (zdb.highlight_document(
             'issue889'::regclass,
             json_build_object('name', name, 'testdate', testdate, 'testval', testval),
             'my_list="2024-04-02" OR my_list="2024-04-09" OR testdate="2024-04-01"')).*,
                                        id as primary_key
                                 FROM matches)
SELECT *
FROM highlights
ORDER BY "primary_key", "field_name", "position";
 field_name | array_index |    term    |  type  | position | start_offset | end_offset |     query_clause      | primary_key 
------------+-------------+------------+--------+----------+--------------+------------+-----------------------+-------------
 testdate   |           0 | 2024-04-01 | <DATE> |        1 |            0 |          0 | testdate="2024-04-01" |           1
 testdate   |           0 | 2024-04-02 | <DATE> |        1 |            0 |          0 | testdate="2024-04-02" |           2
 testdate   |           0 | 2024-04-01 | <DATE> |        1 |            0 |          0 | testdate="2024-04-01" |           3
 testdate   |           0 | 2024-04-01 | <DATE> |        1 |            0 |          0 | testdate="2024-04-01" |           4
 testval    |           0 | 2024-04-09 | word   |        1 |            0 |         10 | testval="2024-04-09"  |           4
(5 rows)

DROP TABLE issue889 CASCADE;
